types/yosensi/YO-PurePro/uplink.js

```diff
@@ -0,0 +1,611 @@
+    });
+    return PAYLOAD_VER;
+}
+
+/**
+ * Decode function compliant with ChirpStack payload formatter
+ *
+ * @param {number} port Port on which data was received
+ * @param {object} bytes Bytes of received payload
+ * @returns Decoded data with warnings/errors, if any, according to the TTN v3 payload formatter, accepted by ChirpStack
+ */
+function decoder(port, bytes) {
+    /**
+     * Protocol version
+     */
+    const protocolVersion = bytes[0] & 0xFF;
+    /**
+     * Object of decoded data
+     */
+    let decoded = {};
+    const data = {};
+    /**
+     * Utility functions
+     */
+    const utils = utilityFunctions();
+
+    if (protocolVersion === supportedPayloadVersions().V2) {
+        decoded = payloadV2Parse(bytes, new Date(), utils);
+    }
+    else {
+        decoded = payload({}, [], ["Unsupported payload. Supported versions: ".concat(utils.getStrFromArrOfObj(supportedPayloadVersions()), " but was: ", protocolVersion)]);
+    }
+
+    /**
+     * Read Data
+     */
+    for(let i=0; i<decoded.data.measurements.length; i++) {
+        switch(decoded.data.measurements[i].typeName){
+            case 'temperature':
+                data.temperature = decoded.data.measurements[i].value;
+                break;
+            case 'relative humidity':
+                data.humidity = decoded.data.measurements[i].value;
+                break;
+            case 'pressure':
+                data.pressure = decoded.data.measurements[i].value;
+                break;
+            case 'illuminance':
+                data.illuminance = decoded.data.measurements[i].value;
+                break;
+            case 'TVOC concentration':
+                data.tvoc = decoded.data.measurements[i].value;
+                break;
+            case 'CO':
+                data.co = decoded.data.measurements[i].value;
+                break;
+            case 'CO2':
+                data.co2 = decoded.data.measurements[i].value;
+                break;
+            case 'sound pressure level':
+                data.sound = decoded.data.measurements[i].value;
+                break;
+            case 'dust concentration':
+                switch(decoded.data.measurements[i].address){
+                    case '02':
+                        data.pm2_5 = decoded.data.measurements[i].value;
+                        break;
+                    case '03':
+                        data.pm4 = decoded.data.measurements[i].value;
+                        break;
+                    case '04':
+                        data.pm10 = decoded.data.measurements[i].value;
+                        break;
+                    default:
+                        data.status = "Error";
+                }
+            break;
+            default:
+                data.status = "Error";
+        }
+    }
+    
+    return {
+        data
+    };
+}
+
+function consume(event) {
+    const payloadData = event.data.payloadHex;
+    const {port} = event.data;
```

Comment:
we recommend using the [prettier Vs Code extension](https://prettier.io/docs/en/editors.html) to align formatting. If you are not using this IDE, you can run `npx prettier ./types/yosensi --write`

Reply:
Thank you for the tip, I will do this right away with the devices I sent and with all the future devices.